behave_django/environment.py

```diff
@@ -17,18 +37,15 @@ def before_scenario(self, context):
 
         Sets up the test case, base_url, and the get_url() utility function.
         """
+        context.__class__ = PatchedContext
+        # Simply setting __class__ directly doesn't work
+        # because behave.runner.Context.__setattr__ is implemented wrongly.
+        object.__setattr__(context, '__class__', PatchedContext)
```

Comment:
@mixxorz I'm happy with the concept, it's a good contribution. I'd really like to see that we get it all right at the first shot. We're pretty close.

- As demonstrated in my comment above, the `base_url` thing can be more concise and pythonic.
- I was undecided about the introduction of the `PatchedContext` class, but now I feel it's is a good thing: It's isolating the `context` patching. But unfortunately it's adding complications (line 40-43).

@avsd What do you think? Wouldn't sticking to simple monkey patching keep things simpler?